libraries/ts-command-line/src/providers/CommandLineParameterProvider.ts

```diff
@@ -478,23 +545,40 @@ export abstract class CommandLineParameterProvider {
         help: argparse.Const.SUPPRESS
       });
     }
-
-    this._parameters.push(parameter);
-    this._parametersByLongName.set(parameter.longName, parameter);
   }
 
   private _generateKey(): string {
     return 'key_' + (CommandLineParameterProvider._keyCounter++).toString();
   }
 
+  private _generateScopedLongName(parameter: CommandLineParameter): string {
+    if (!parameter.parameterScope) {
+      throw new Error(`Cannot generate scoped long name for parameter without scope "${parameter.longName}"`);
+    }
+
+    // Parameter long name is guranteed to start with '--' since this is validated in the
+    // CommandLineParameter constructor.
+    const unprefixedLongName: string = parameter.longName.slice(2);
+    return `--${parameter.parameterScope}:${unprefixedLongName}`;
```

Comment:
Could we expose a public API that does the string transformations of `("--thing", "scope") --> "--scope:thing"`? (And maybe also `"--scope:thing" --> ("--thing", "scope")`?)  We wouldn't want callers writing their own code that tries to parse/format this syntax.